### PR TITLE
Convert bin to cmd

### LINKED ISSUE #499: split apart cmd into submodules
cmd folder is becoming a bit cluttered. best to split apart and pull in using _ imports in the main.go file at root.
use the [ExecuteContext](https://godoc.org/github.com/spf13/cobra#Command.ExecuteContext) command to pass context around

### LINKED ISSUE #500: bring vault bin files into cmd structure
to start we need
* vault-import-keys
* vault-sign-settlement

to get everything under a single binary we need all bin files under cmd
secondary bin files include
* vault-create-wallet
* vault-init
* vault-unseal
* vault-transfer-funds

### PR DESCRIPTION
**Pull request by maikelmclauflin**
### Summary
this refactor pulls all necessary bits into a single output binary to be used on the offline settlement tooling and converts former bin files into cmd (cobra) files

fixes #500 
fixes #499 

### Type of change ( select one )

- [x] Product feature
- [ ] Bug fix
- [ ] Performance improvement
- [ ] Refactor
- [ ] Other

### Tested Environments

- [ ] Development
- [ ] Staging
- [ ] Production

### Before submitting this PR:

- [ ] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security / privacy review if needed
- [ ] Have you performed a self review of this PR?

### CODE REVIEW

File: settlement/settlement.go [hunk 1]
```diff
@@ -50,6 +50,8 @@ type Transaction struct {
 
 // AntifraudTransaction a transaction object that comes from antifraud
 type AntifraudTransaction struct {
+	// CreatedAt         time.Time       `json:"created_at"`
```
Comment (husobee) [inline settlement/settlement.go:53]: Are these comments valid?

File: utils/clients/client.go [hunk 2]
```diff
@@ -86,6 +86,8 @@ func NewWithProxy(name string, serverURL string, authToken string, proxyURL stri
 			Transport: middleware.InstrumentRoundTripper(
 				&http.Transport{
 					Proxy: proxy,
+					// can i just remove this?
+					// DialTLS: pindialer.MakeDialer(upholdCertFingerprint),
```
Comment (husobee) [inline utils/clients/client.go:90]: Yes, this is only valid for the uphold client afaik, which is created here: https://github.com/brave-intl/bat-go/blob/master/utils/wallet/provider/uphold/uphold.go#L120

File: utils/context/keys.go [hunk 3]
```diff
@@ -32,6 +32,8 @@ const (
 	RatiosCacheExpiryDurationCTXKey CTXKey = "ratios_client_cache_expiry"
 	// RatiosCachePurgeDurationCTXKey - context key for ratios client cache purge
 	RatiosCachePurgeDurationCTXKey CTXKey = "ratios_client_cache_purge"
+	// LogEvent holds a log event to be passed through settlement functions
+	LogEvent CTXKey = "log_event"
```
Comment (husobee) [inline utils/context/keys.go:36]: this log event is a zerolog log event? is there a reason we are not passing a sub logger instead?

**Review by husobee — APPROVED** [review]
looks good.  Much cleaner than the flat structure.  Only a few comments inline.

File: settlement/settlement.go (hunk 1 above)
Comment (maikelmclauflin) [reply settlement/settlement.go:53]: no. looks like the last changes were clobbered.

File: utils/clients/client.go (hunk 2 above)
Comment (maikelmclauflin) [reply utils/clients/client.go:90]: super. removed

File: utils/context/keys.go (hunk 3 above)
Comment (maikelmclauflin) [reply utils/context/keys.go]: fixed in 7f0ed18

File: utils/clients/client.go (hunk 2 above)
Comment (maikelmclauflin) [reply utils/clients/client.go]: fixed in 7f0ed18

File: settlement/settlement.go (hunk 1 above)
Comment (maikelmclauflin) [reply settlement/settlement.go]: fixed in 7f0ed18

File: cmd/settlement/paypal.go [hunk 4]
```diff
@@ -229,9 +227,10 @@ func PaypalWriteTransactions(outPath string, metadata *[]settlement.Transaction)
 }
 
 // PaypalWriteMassPayCSV writes a csv for using with Paypal web mass payments
-func PaypalWriteMassPayCSV(outPath string, metadata *[]paypal.Metadata) error {
+func PaypalWriteMassPayCSV(ctx context.Context, outPath string, metadata *[]paypal.Metadata) error {
 	rows := []*paypal.MassPayRow{}
 	total := decimal.NewFromFloat(0)
+	logEvent := ctx.Value(appctx.LogEvent).(*zerolog.Event)
```
[the remark this answers is not in the document]
Comment (maikelmclauflin) [reply cmd/settlement/paypal.go]: fixed in dbf6617

**Review by husobee — APPROVED** [review]
lgtm

**Pull request merged**